iroh-api/src/api.rs

```diff
@@ -37,23 +40,46 @@ pub trait Api {
     async fn watch<'a>(&self) -> LocalBoxStream<'a, StatusTable>;
 }
 
-impl<'a> Iroh<'a> {
-    pub fn new(client: &'a Client) -> Self {
+impl Iroh {
+    pub async fn new(
+        config_path: Option<&Path>,
+        overrides_map: HashMap<String, String>,
+    ) -> Result<Self> {
+        let cfg_path = iroh_config_path(CONFIG_FILE_NAME)?;
```

Comment:
This is not new code, but just moved around, correct?

Reply:
Correct, from iroh-ctl